src/css/property-descriptors/font-family.ts

```diff
@@ -33,6 +33,6 @@ export const fontFamily: IPropertyListDescriptor<FontFamily> = {
         if (accumulator.length) {
             results.push(accumulator.join(' '));
         }
-        return results.map((result) => (result.indexOf(' ') === -1 ? result : `'${result}'`));
+        return results.map((result) => (result.indexOf(' ') === -1 && result.match(/^[a-zA-Z]/) ? result : `'${result}'`));
```

Comment:
LGTM, but I think we should be even more strict.
Looking at [this article](https://developer.mozilla.org/en-US/docs/Web/CSS/font-family), I see this:

> Font family names must either be given quoted as strings, or unquoted as a sequence of one or more identifiers. This means that punctuation characters and digits at the start of each token must be escaped in unquoted font family names.
>
> It is a good practice to quote font family names that contain white space, digits, or punctuation characters other than hyphens.

So it looks like you're correct, but interestingly in other translations of this article, and also in their own examples, they say that ALL these examples are invalid:
```
font-family: Goudy Bookletter 1911, sans-serif;
font-family: Red/Black, sans-serif;
font-family: "Lucida" Grande, sans-serif;
font-family: Ahem!, sans-serif;
font-family: test@foo, sans-serif;
font-family: #POUND, sans-serif;
font-family: Hawaii 5-0, sans-serif;
```

So, really, we should only keep font-family unescaped if it only consists of A-Z and hyphens. So, probably, this is a more correct thing to do:

```
return results.map((result) => (result.match(/^[a-z][a-z-]*$/i) ? result : `'${result}'`));
```

Thoughts?

Reply:
@8bitjoey 
About [CSS identifiers](https://www.w3.org/TR/CSS2/syndata.html#value-def-identifier)
> In CSS, identifiers (including element names, classes, and IDs in selectors) can contain only the characters [a-zA-Z0-9] and ISO 10646 characters U+00A0 and higher, plus the hyphen (-) and the underscore (_); they cannot start with a digit, two hyphens, or a hyphen followed by a digit. Identifiers can also contain escaped characters and any ISO 10646 character as a numeric code (see next item). For instance, the identifier "B&W?" may be written as "B\&W\?" or "B\26 W\3F".

So `-abc` `ab123`are valid.
Thought I would just need to fix our use case, which is to use guid as font family name, to keep it simple.

Reply:
It's actually a bit more complex that that. I agree that `-abc` is valid, but then `_abc` should be valid too (since it complies [ident-token](https://www.w3.org/TR/css-syntax-3/#ident-token-diagram) syntax).

Looking into [W3C definition of <family-name>](https://www.w3.org/TR/css-fonts-4/#family-name-syntax), all these examples are still invalid:
```
font-family: Red/Black, sans-serif;
font-family: "Lucida" Grande, sans-serif;
font-family: Ahem!, sans-serif;
font-family: test@foo, sans-serif;
font-family: #POUND, sans-serif;
font-family: Hawaii 5-0, sans-serif;
```

Although, for example, the "test@foo" does start with a letter and doesn't have spaces in it. So current code will leave it unquoted. And otherwise, current code will quote `-abc`, although it's a valid name.

So, I guess we have these options:
1. be nerdy and write regex which will comply with all font-family parsing rules. Too hard imo and not really worth it.
2. be paranoid and escape everything. But then we need to leave [generic font-families](https://www.w3.org/TR/css-fonts-4/#generic-font-families) unescaped. Which makes this option not worth it too.
3. keep current solution. I personally am fine with it. It catches almost all cases that need to be escaped.
4. Be just a tiny bit more smart (ignoring non-ASCII and escape characters comparing this to option #1):
```
return results.map((result) => (result.match(/^-?[a-z_][a-z0-9_-]*$/i) ? result : `'${result}'`));
```

Reply:
Again, I fine with current solution. Just though that we can make it even more better since we're improving it already.

Reply:
@8bitjoey I'd like to be `a tiny bit more smart`. Updated.